src/App.tsx

```diff
@@ -1,61 +1,161 @@
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function findMaxId(items: Title[]) {
+  let maxId = 0;
+
+  items.forEach(item => {
+    if (item.id > maxId) {
+      maxId = item.id;
+    }
+  });
+
+  return maxId;
+}
+
+type Title = {
+  id: number,
+  title: string,
+  completed: boolean,
+  userId: number,
+};
+
+export const App: React.FC = () => {
+  const [newTodo, setNewTodo] = useState<Title>({
+    id: 0,
+    title: '',
+    completed: false,
+    userId: 0,
+  });
+
+  const [fieldErrors, setFieldErrors] = useState<{ [key: string]: string }>({});
+
+  const [todos, setTodos] = useState<Todo[]>(() => todosFromServer
+    .map((todo) => ({
+      ...todo,
+      user: getUser(todo.userId),
+    })));
+
+  const [selectedUserId, setSelectedUserId] = useState({
+    select: 0,
+  });
+
+  const handleSelected = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const { name, value } = e.target;
+
+    setSelectedUserId(prevState => ({
+      ...prevState,
+      [name]: +value,
+    }));
+    setFieldErrors({
+      ...fieldErrors,
+      select: value === '0' ? 'Please choose a user' : '',
+    });
+  };
+
+  const handleTitle = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodo({
+      ...newTodo,
+      title: e.target.value,
+    });
+    setFieldErrors({
+      ...fieldErrors,
+      title: newTodo.title.trim() === '' ? 'Please enter a title' : '',
+    });
+  };
+
+  const handleSubmit: FormEventHandler<HTMLFormElement> = (
+    e: React.FormEvent<HTMLFormElement>,
+  ) => {
+    e.preventDefault();
+    const titleError = newTodo.title.trim() === ''
+      ? 'Please enter a title' : '';
+    const selectError = selectedUserId.select === 0
+      ? 'Please choose a user' : '';
+
+    setFieldErrors({
+      title: titleError,
+      select: selectError,
+    });
+    if (titleError === '' && selectError === '') {
+      const maxId = findMaxId(todos);
+      const nextId = maxId + 1;
+      const newTodoItem = {
+        ...newTodo, id: nextId, userId: selectedUserId.select,
+      };
+
+      setNewTodo(newTodoItem);
+
+      const newTodos = [...todos, {
+        ...newTodoItem, user: getUser(newTodoItem.userId),
+      }];
+
+      setTodos(
+        newTodos,
+      );
+      setNewTodo({
+        id: 0,
+        title: '',
+        completed: false,
+        userId: 0,
+      });
+      setSelectedUserId({
+        select: 0,
+      });
+    }
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form action="/api/users" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            data-cy="titleInput"
+            value={newTodo.title}
```

Comment:
```suggestion
            value={newTodo.title}
            placeholder="Please enter a title"
```
![Screenshot from 2023-04-02 15-16-36](https://user-images.githubusercontent.com/33029864/229352323-05a2aa99-9551-43c7-a34e-40e78d661293.png)